### PR TITLE
Run python unittests; configure codecov via YAML

### PR DESCRIPTION
**Pull request by mikegray**
fixing #2509, adding recommended YAML file for better codecov.io integration

### CODE REVIEW

File: codecov.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+coverage:
+  precision: 2         # how many decimal places to display in the UI: 0 <= value <= 4
+  round: down          # how coverage is rounded: down/up/nearest
+  range: 60...100      # custom range of coverage colors from red -> yellow -> green
+
+  notify:
+    slack:
+      default:
+        url: "secret:n+BYhIXTXsaCiMKB3vOf6yP68ytdKd3WpXtJFWPEUsEWXDiGnU5dTB5DO2yv8tR0COdxvs7K31hVpEfHEXdoXOaQhUw3FKf3fh8KZDLN7CGTbeDhw1uNGGyBr2d2TWnopzYtcXomdwMmuckARtiWQx0YXJiZY9YyCrIoDK9HIJQ="
```
Comment (arv) [inline codecov.yml]: is this ok to keep in git?

Comment (mikegray) [inline codecov.yml]: Yes, I generated it via the codecov.io website so only they know how to decrypt it to determine the real URL to use.

**Comment by mikegray** [comment]
What about these failing python unit tests though?  It appears to work fine on my Mac, so maybe it is not deterministic? or ordering is based on something different for Linux vs. Darwin?

**Comment by arv** [comment]
Maybe we need to sort these or use `set()`/`frozenset()`?

**Comment by mikegray** [comment]
@arv PTAL, using sorted instead of frozenset. unittest2 package has `assertElementsEqual` but we aren't using that. Python 3.2+ unittest has another similar option, but we are using Python 2.7.

**Comment by arv** [comment]
Still LGTM

**Pull request merged**